presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -233,6 +233,7 @@ private void processNewSources()
         long maxRuntime = duration.roundTo(TimeUnit.NANOSECONDS);
 
         Optional<ListenableFuture<?>> result = tryWithLock(100, TimeUnit.MILLISECONDS, () -> {
+            driverContext.getYieldSignal().setWithDelay(maxRuntime, driverContext.getYieldExecutor());
```

Comment:
with some changes we can make the driver yield signal closeable and use it like:

```
try (DriverYieldSignal ignored = driverContext.getYieldSignal().setWithDelay(maxRuntime, driverContext.getYieldExecutor())) {
    driverContext.startProcessTimer();
    long start = System.nanoTime();
    do {
        ListenableFuture<?> future = processInternal();
        if (!future.isDone()) {
            return wakeUpOnRevokeRequest(future);
        }
    }
    while (System.nanoTime() - start < maxRuntime && !isFinishedInternal());
}
finally {
    driverContext.recordProcessed();
}
```